shims/spark311/src/main/scala/com/nvidia/spark/rapids/shims/spark311/Spark311Shims.scala

```diff
@@ -276,12 +276,20 @@ class Spark311Shims extends Spark301Shims {
         }),
       GpuOverrides.exec[InMemoryTableScanExec](
         "Implementation of InMemoryTableScanExec to use GPU accelerated Caching",
-        ExecChecks(TypeSig.commonCudfTypes + TypeSig.DECIMAL, TypeSig.all),
+        ExecChecks((TypeSig.commonCudfTypes + TypeSig.DECIMAL + TypeSig.STRUCT).nested()
+          .withPsNote(TypeEnum.DECIMAL,
+            "Negative scales aren't supported at the moment even with " +
+              "spark.sql.legacy.allowNegativeScaleOfDecimal set to true. This is because Parquet " +
```

Comment:
Is there a follow on issue for this? Probably not that important. We could hack around it, but it is probably not worth it to do so.

Reply:
I have added it to this issue https://github.com/NVIDIA/spark-rapids/issues/1143